### PR TITLE
Feature/us 18/edit user profile

### PR DESCRIPTION
**Pull request by Karvlox**
What did I do?
Implement the user profile editing functionality, where you can edit the CI, Name, LastName, Email, password and gender.
This was done so that the user can edit these fields which will be used for their orders later.

Why did I do it?
Because the users usually modify their data for personal reasons so it is something important in this area because they will use their data to place their orders.

How did I do it?
I communicated with the API created by the team of "Los Jailes" which communicates with our DB which gets the data from an account that is logged in and can be updated once the necessary changes have been made as long as the validations are passed.

Author: @Karvlox 
Code-Reviewer: @Misax148 
Tester: @victor-villca

### CODE REVIEW

File: src/app/pages/user-profile/UserProfile.css [hunk 1]
```diff
@@ -82,6 +82,35 @@ body {
   appearance: none;
 }
 
+.button-edit {
+  text-align: center; /* Centra el contenido del botón */
```
Comment (Misax148) [inline src/app/pages/user-profile/UserProfile.css]: Delete comments please, unless it is important to clarify something relevant.

File: src/app/pages/user-profile/UserProfile.css [hunk 2]
```diff
@@ -82,6 +82,35 @@ body {
...
+  border-radius: 20px;
+  color: #fff;
+  font-size: 16px;
+  padding: 5px;
+  cursor: pointer;
+  transition: .3s;
+}
+
+.button-edit span {
+  display: inline-block; /* Añade un display inline-block para mantener el tamaño original */
```
Comment (Misax148) [inline src/app/pages/user-profile/UserProfile.css]: Delete comments please, unless it is important to clarify something relevant.

File: src/app/pages/user-profile/page.jsx [hunk 3]
```diff
@@ -1,47 +1,146 @@
...
+    } catch (error) {
+      console.error("Error updating user data:", error);
+    }
+  };
+
+
+  const getUser = async () => {
+    try {
+      const u = await api.get('/User/email/' + session.data.user.email);
+      console.log(u);
```
Comment (Misax148) [inline src/app/pages/user-profile/page.jsx]: Deletes messages in console.

File: src/app/pages/user-profile/page.jsx [hunk 4]
```diff
@@ -1,47 +1,146 @@
...
+      return u.data;
+    } catch (error) {
+      console.error("Error fetching user:", error);
```
Comment (Misax148) [inline src/app/pages/user-profile/page.jsx:123]: Delete this message in the console and replace it with an Error.

File: src/app/pages/user-profile/page.jsx [hunk 5]
```diff
@@ -1,47 +1,146 @@
...
+    e.preventDefault();
+    try {
+      const [isEmailValid, emailErrorMessage] = validateEmail(email);
+      const [isPasswordValid, passwordErrorMessage] = validatePassword(password);
+      const [isCiValid, ciErrorMessage] = validateCiField(ci);
+      const [isNameValid, nameErrorMessage] = validateTextField(name, 'Name');
+      const [isLastNameValid, lastNameErrorMessage] = validateTextField(lastName, 'Last Name');
+
+      if (!isEmailValid) {
+        console.error('Email Validation Error:', emailErrorMessage);
```
Comment (Misax148) [inline src/app/pages/user-profile/page.jsx]: Throws user-friendly errors, or changes these errors to throw exceptions such as an `Error`.

File: src/app/pages/user-profile/page.jsx [hunk 6]
```diff
@@ -1,47 +1,146 @@
...
+        return;
+      }
+      if (!isPasswordValid) {
+        console.error('Password Validation Error:', passwordErrorMessage);
```
Comment (Misax148) [inline src/app/pages/user-profile/page.jsx]: Throws user-friendly errors, or changes these errors to throw exceptions such as an `Error`.

File: src/app/pages/user-profile/page.jsx [hunk 7]
```diff
@@ -1,47 +1,146 @@
...
+        return;
+      }
+      if (!isCiValid) {
+        console.error('CI Validation Error:', ciErrorMessage);
```
Comment (Misax148) [inline src/app/pages/user-profile/page.jsx]: Throws user-friendly errors, or changes these errors to throw exceptions such as an `Error`.

File: src/app/pages/user-profile/page.jsx [hunk 8]
```diff
@@ -1,47 +1,146 @@
...
+        return;
+      }
+      if (!isNameValid) {
+        console.error('Name Validation Error:', nameErrorMessage);
```
Comment (Misax148) [inline src/app/pages/user-profile/page.jsx]: Throws user-friendly errors, or changes these errors to throw exceptions such as an `Error`.

File: src/app/pages/user-profile/page.jsx [hunk 9]
```diff
@@ -1,47 +1,146 @@
...
+        return;
+      }
+      if (!isLastNameValid) {
+        console.error('Last Name Validation Error:', lastNameErrorMessage);
```
Comment (Misax148) [inline src/app/pages/user-profile/page.jsx]: Throws user-friendly errors, or changes these errors to throw exceptions such as an `Error`.

File: src/app/pages/user-profile/UserProfile.css (hunk 1 above)
Comment (Karvlox) [reply src/app/pages/user-profile/UserProfile.css]: Done.

File: src/app/pages/user-profile/page.jsx (hunk 3 above)
Comment (Karvlox) [reply src/app/pages/user-profile/page.jsx]: Done.

File: src/app/pages/user-profile/page.jsx (hunk 5 above)
Comment (Karvlox) [reply src/app/pages/user-profile/page.jsx]: They are like exceptions to show in console.

File: src/app/pages/user-profile/page.jsx (hunk 7 above)
Comment (Karvlox) [reply src/app/pages/user-profile/page.jsx]: They are like exceptions to show in console.

File: src/app/pages/user-profile/page.jsx (hunk 9 above)
Comment (Karvlox) [reply src/app/pages/user-profile/page.jsx]: They are like exceptions to show in console.

**Review by Misax148 — APPROVED** [review]

**Review by victor-villca — CHANGES_REQUESTED** [review]
The user should see its password, and not the encrypted one
image

**Review by victor-villca — CHANGES_REQUESTED** [review]
Please, make the text does not hit the icon, you can do it by making the password text smaller or with overflow hidden property.
image

**Review by victor-villca — CHANGES_REQUESTED** [review]
I found a few bugs:
- I can put special characters or numbers to all text inputs, like the lastname or name. 
- I can change the email and password, which should not be possible, due to the email is unique and the password is shown encypted. This should not be allowed.
- As it can be seen in the video, the new added info was not updated/saved in th DB.

[Screencast from 2023-11-14 07-04-30.webm](https://github.com/Les-Jailes/Boutique_Clothing/assets/101595370/811563b6-f0b2-4ca1-817b-9d16dd5d410e)

**Comment by Karvlox** [comment]
A team discussion was held and the problems presented were solved.

**Review by victor-villca — APPROVED** [review]
I see that the new commits have fixed my revisions. Approved.

**Pull request merged**